core/src/main/antlr/xtdb/antlr/Sql.g4

```diff
@@ -226,6 +232,7 @@ exprPrimary
     | exprPrimary '.' fieldName #FieldAccess
     | exprPrimary '[' expr ']' #ArrayAccess
     | exprPrimary '::' dataType #PostgresCastExpr
+    | exprPrimary '::' schemaName '.' 'REGTYPE' '::' schemaName '.' 'TEXT' # PgCatalogResolveOid
```

Comment:
this looks like two distinct expressions? i.e. schemaName.regtype and `expr :: schema.text`?

in fact, the latter looks like it's standard cast syntax, so do we need to make that change there instead?

Reply:
> Also - making the cast expression recursive would complicate it w/o much benefit?

having two expressions in one rule is more complicated, we don't do this anywhere else - in a grammar, you decompose the syntax down into its constituent parts.

> happy to be told otherwise if you have a strong feeling about this.

yes please :pray: